independent-projects/tools/devtools-common/src/main/java/io/quarkus/cli/plugin/PluginMangerState.java

```diff
@@ -173,8 +174,30 @@ public Map<String, Plugin> extensionPlugins() {
         Map<String, Plugin> extensionPlugins = new HashMap<>();
         projectRoot.map(r -> quarkusProject.get()).ifPresent(project -> {
             try {
-                Set<ArtifactKey> installed = project.getExtensionManager().getInstalled().stream()
-                        .map(ArtifactCoords::getKey).collect(Collectors.toSet());
+                Set<ArtifactKey> installed = new HashSet<>();
+                // Direct extensions
+                for (ArtifactCoords coords : project.getExtensionManager().getInstalled()) {
+                    installed.add(coords.getKey());
+                }
+                // Transitives
+                for (Extension e : project.getExtensionsCatalog().getExtensions()) {
+                    if (!installed.contains(e.getArtifact().getKey())) {
```

Comment:
Here the `installed` set is probably assumed to be direct dependencies but it changes as you iterate through the catalog, since transitive dependency keys are added to it and may be treated as direct ones while the iteration goes on.

Reply:
Good point